chia/cmds/rpc.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+import json
+import sys
+from typing import Any, Dict, List, Optional, TextIO
+
+import click
+from aiohttp import ClientResponseError
+
+from chia.util.config import load_config
+from chia.util.default_root import DEFAULT_ROOT_PATH
+from chia.util.ints import uint16
+
+services: List[str] = ["farmer", "full_node", "harvester", "wallet"]
+
+
+async def call_endpoint(
+    service: str, host: str, port: uint16, endpoint: str, request: Dict[str, Any], config: Dict[str, Any]
+) -> Dict[str, Any]:
+    from chia.rpc.rpc_client import RpcClient
+
+    try:
+        client = await RpcClient.create(host, port, DEFAULT_ROOT_PATH, config)
+    except Exception as e:
+        raise Exception(f"Failed to create RPC client {service}: {e}")
+    result: Dict[str, Any]
+    try:
+        result = await client.fetch(endpoint, request)
+    except ClientResponseError as e:
+        if e.code == 404:
+            raise Exception(f"Invalid endpoint for {service}: {endpoint}")
+        raise e
+    except Exception as e:
+        raise Exception(f"Request failed: {e}")
+    finally:
+        client.close()
+        await client.await_closed()
+    return result
+
+
+def print_result(json_dict: Dict[str, Any]) -> None:
+    print(json.dumps(json_dict, indent=4, sort_keys=True))
+
+
+def get_routes(service: str, config: Dict[str, Any]) -> Dict[str, Any]:
+    return asyncio.run(
+        call_endpoint(service, config["self_hostname"], uint16(config[service]["rpc_port"]), "get_routes", {}, config)
+    )
+
+
+@click.group("rpc", short_help="RPC Client")
+def rpc_cmd() -> None:
+    pass
+
+
+@rpc_cmd.command("endpoints", help="Print all endpoints of a service")
+@click.argument("service", type=click.Choice(services))
+def endpoints_cmd(service: str) -> None:
+    config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
+    try:
+        routes = get_routes(service, config)
+        for route in routes["routes"]:
+            print(route[1:])
+    except Exception as e:
+        print(e)
+
+
+@rpc_cmd.command("status", help="Print the status of all available RPC services")
+def status_cmd() -> None:
+    config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
+
+    def print_row(c0: str, c1: str) -> None:
+        c0 = "{0:<12}".format(f"{c0}")
+        c1 = "{0:<9}".format(f"{c1}")
+        print(f"{c0} | {c1}")
+
+    print_row("SERVICE", "STATUS")
+    print_row("------------", "---------")
+    for service in services:
+        status = "ACTIVE"
+        try:
+            if not get_routes(service, config)["success"]:
+                raise Exception()
+        except Exception:
+            status = "INACTIVE"
+        print_row(service, status)
+
+
+def create_commands() -> None:
+    for service in services:
```

Comment:
Dynamically executing `def` like this is not working because the final `def func_name` will always override the previous `def func_name`. (Maybe it works if function names are different.)
You can try ` chia rpc full_node get_blockchain_state` to check this.
This works if  
`services: List[str] = ["farmer", "full_node", "harvester", "wallet"]`  
is updated to  
`services: List[str] = ["farmer", "harvester", "wallet", "full_node"]`

Reply:
Not sure what you mean here? `chia rpc full_node get_blockchain_state` works fine for me.

Reply:
After I force-pulled your latest commit and error has gone...
What I meant here is repeating `def func_name` with the same `func_name` discards function definition except for the last one.
But you seem like adding `service` variable into the decorators so it eventually defines unique func name per a `service` so error went away.
This has been resolved.